cfe_internal/enterprise/CFE_knowledge.cf

```diff
@@ -34,93 +34,114 @@ bundle agent cfe_internal_setup_knowledge
       "$(cfe_internal_hub_vars.docroot)"
       comment => "All files in there should be at least 0644",
       handle => "cfe_internal_setup_knowledge_files_doc_root_2",
-      perms => m("0644"),
+      file_select => cfe_internal_docroot_perms,
+      perms => mog("0440", "root", $(def.cf_apache_group) ),
       depth_search => recurse_exclude("inf");   # see exclude dirs in recurse_exclude() body
 
       "$(cfe_internal_hub_vars.docroot)/.htaccess"
       comment => "Correct up htaccess file in doc root",
       handle => "cfe_internal_setup_knowledge_files_doc_root_htaccess",
-      perms => mog("0644","root","root"),
       copy_from => no_backup_cp("$(sys.workdir)/share/GUI/Apache-htaccess");
 
       "$(cfe_internal_hub_vars.docroot)/hub/."
       comment => "Create a necessary folder for cf-hub",
       handle => "cfe_internal_setup_knowledge_files_doc_root_hub",
       create => "true",
       depth_search => recurse_basedir("inf"),
-      perms => mog("0755","root","root");
+      perms => mog("0440","root", $(def.cf_apache_group) );
 
       "$(cfe_internal_hub_vars.docroot)/scripts/."
       comment => "Ensure permissions for $(cfe_internal_hub_vars.docroot)/scripts",
       handle => "cfe_internal_setup_knowledge_files_doc_root_scripts",
       create => "true",
       depth_search => recurse_basedir("inf"),
-      perms => mog("0644","root","root");
+      perms => mog("0440", "root", $(def.cf_apache_group) );
 
       "$(cfe_internal_hub_vars.docroot)/tmp/."
       handle => "cfe_internal_setup_knowledge_files_doc_root_tmp",
       create => "true",
       depth_search => recurse_basedir("inf"),
-      perms => mog("0644",$(def.cf_apache_user),$(def.cf_apache_group)),
+      perms => mog("0660", "root", $(def.cf_apache_group)),
       comment => "Ensure permissions for $(cfe_internal_hub_vars.docroot)/tmp.
                   This is where scheduled reports generated by Mission Portal
                   (temp files to email)";
 
       "$(cfe_internal_hub_vars.docroot)/application/logs/."
-      comment => "Make sure log folder has been created and has right permisions",
+      comment => "Only the webserver needs to be able to write to application logs.",
       handle => "cfe_internal_setup_knowledge_files_doc_root_application_logs",
       create => "true",
       depth_search => recurse_basedir("inf"),
       file_select => cfe_internal_exclude_index_html,
-      perms => mog("0644",$(def.cf_apache_user),$(def.cf_apache_group));
+      perms => mog("0440", "root", $(def.cf_apache_group) );
```

Comment:
ok, better to err on the safe side. :+1: 

Too long since I've looked at Unix perms, but pls make sure that root has access so they don't need to su apache or change the perms to look at the logs.